pkg/errors/armerrors.go

```diff
@@ -15,12 +15,81 @@ limitations under the License.
 package errors
 
 import (
+	"bytes"
+	"encoding/json"
 	"errors"
+	"fmt"
+	"io"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 )
 
+type CreationResponseError struct {
+	*azcore.ResponseError
+}
+
+// TODO - this is a very WIP dummy implementation, trying to gather thoughts first before I make it more robust
+func (e *CreationResponseError) Error() string {
+	msg := &bytes.Buffer{}
+	if e.RawResponse != nil {
+		fmt.Fprintf(msg, "RESPONSE %d: %s\n", e.RawResponse.StatusCode, e.RawResponse.Status)
+	} else {
+		fmt.Fprintln(msg, "RESPONSE: Unknown (Missing RawResponse)")
+	}
+
+	if e.ErrorCode != "" {
+		fmt.Fprintf(msg, "ERROR CODE: %s\n", e.ErrorCode)
+	} else {
+		// TODO - whats the difference between this code and the one in response body
+		fmt.Fprintln(msg, "ERROR CODE: UNAVAILABLE")
+	}
+
+	if e.RawResponse != nil {
+		body, err := io.ReadAll(e.RawResponse.Body)
+		if err != nil {
+			fmt.Fprintln(msg, "Error reading response body:", err)
+		}
+
+		if len(body) > 0 {
+			if err := json.Indent(msg, body, "", "  "); err != nil {
+				// failed to pretty-print so just dump it verbatim
+				fmt.Fprint(msg, string(body))
+			}
+			// the standard library doesn't have a pretty-printer for XML
+			fmt.Fprintln(msg)
+		} else {
+			fmt.Fprintln(msg, "Response contained no body")
+		}
+
+		e.RawResponse.Body.Close()
+	}
+
+	// After showing the important information, add request details
+	if e.RawResponse != nil && e.RawResponse.Request != nil {
+		fmt.Fprintf(msg, "REQUEST: %s %s://%s%s\n",
+			e.RawResponse.Request.Method,
+			e.RawResponse.Request.URL.Scheme,
+			e.RawResponse.Request.URL.Host,
+			e.RawResponse.Request.URL.Path)
+	} else {
+		fmt.Fprintln(msg, "REQUEST: Information not available")
+	}
+
+	return msg.String()
+}
+
+func (e *CreationResponseError) Unwrap() error {
+	return e.ResponseError
```

Comment:
This is probably good to support but just note that it means if the caller does `errors.As` looking for an `azcore.ResponseError` it'll unwrap right past this wrapper and give the longer `Error` response instead of our shortened/rearranged one.

Reply:
Did you drop this unwrap?

can you talk about why? I see there are some risks with unwrapping "past" the wrapper, but it still might be OK as long as the caller is aware of that? How often does the caller actually As(ResponseError) and isn't wrapping without Unwrap gonna break that anyway?

Reply:
As discussed offline, we will add back the `Unwrap` method. It is fine if user unwraps past our custom wrapper. Idea is that the user would be explicit about wanting to use the wrapper or not (if they care about the message formatting)